shared/protocol/src/main/java/io/pravega/shared/protocol/netty/CommandEncoder.java

```diff
@@ -212,6 +213,7 @@ protected void encode(ChannelHandlerContext ctx, Object msg, ByteBuf out) throws
                     continueAppend(data, out);
                     if (bytesLeftInBlock == 0) {
                         completeAppend(null, out);
+                        flushRequired();
```

Comment:
We have access to `ChannelHandlerContext ctx ` here. Invoking `ctx.flush()` has the same effect too.

Reply:
I thought that too at first. Unfortunately the super class calls this method, then it calls write afterwards. So if we call flush here, it won't flush the data that was serialized here, but only whatever is already waiting. That is why I added the super class so that it can call the method after the invocation has returned. 

If we were to re-write all of the MessageToByteEncoder here, it could be done in a much simpler way, but I didn't want to pull in all that code and further muddle the logic of this class, which is already too complicated.